packages/twenty-front/src/modules/workflow/workflow-steps/components/WorkflowRunStepInputDetail.tsx

```diff
@@ -0,0 +1,47 @@
+import { JsonNestedNode } from '@/workflow/components/json-visualizer/components/JsonNestedNode';
+import { useWorkflowRun } from '@/workflow/hooks/useWorkflowRun';
+import { useWorkflowRunIdOrThrow } from '@/workflow/hooks/useWorkflowRunIdOrThrow';
+import { getWorkflowRunStepContext } from '@/workflow/workflow-steps/utils/getWorkflowRunStepContext';
+import styled from '@emotion/styled';
+import { isDefined } from 'twenty-shared';
+import { IconBrackets } from 'twenty-ui';
+
+const StyledContainer = styled.div`
+  padding-block: ${({ theme }) => theme.spacing(4)};
+  padding-inline: ${({ theme }) => theme.spacing(3)};
+`;
+
+export const WorkflowRunStepInputDetail = ({ stepId }: { stepId: string }) => {
+  const workflowRunId = useWorkflowRunIdOrThrow();
+  const workflowRun = useWorkflowRun({ workflowRunId });
+
+  if (
+    !(
+      isDefined(workflowRun) &&
+      isDefined(workflowRun.context) &&
+      isDefined(workflowRun.output?.flow)
+    )
+  ) {
+    return null;
+  }
+
+  const stepContext = getWorkflowRunStepContext({
+    context: workflowRun.context,
+    flow: workflowRun.output.flow,
+    stepId,
+  });
+
+  return (
+    <StyledContainer>
+      <JsonNestedNode
+        elements={stepContext.map(({ id, name, context }) => ({
+          id,
+          label: name,
+          value: context,
+        }))}
```

Comment:
I don't directly use the `<JsonTree />` component because I must support several steps with the same name. If I used `<JsonTree />`, I would have to pass an object and it couldn't handle several steps having the same name (the name would be the key in the object).

I use the `<JsonNestedNode />` component as it lets me choose the label and the value of each element.